src/storage/txn/actions/flashback_to_version.rs

```diff
@@ -490,34 +514,71 @@ pub mod tests {
 
         let flashback_start_ts = *ts.incr();
         // Rollback nothing.
-        assert_eq!(
-            must_rollback_lock(&mut engine, k, ts, flashback_start_ts),
-            0
-        );
+        assert_eq!(must_rollback_lock(&mut engine, k, flashback_start_ts), 0);
         // Lock and write the value of `k`.
         assert_eq!(
             must_prewrite_flashback_key(&mut engine, k, 2, flashback_start_ts),
             2
         );
+        // Retry Prepare
         // Unlock `k`, put rollback record and delete the value of `k`.
-        assert_eq!(
-            must_rollback_lock(&mut engine, k, ts, flashback_start_ts),
-            3
-        );
+        assert_eq!(must_rollback_lock(&mut engine, k, flashback_start_ts), 3);
         // Lock and write the value of `k`.
         assert_eq!(
             must_prewrite_flashback_key(&mut engine, k, 2, flashback_start_ts),
             2
         );
+        // Retry Prepare
         // Only unlock `k` since there is an overlapped rollback record.
+        assert_eq!(must_rollback_lock(&mut engine, k, flashback_start_ts), 1);
+        // Only lock `k` since the value of `k` has already existed.
         assert_eq!(
-            must_rollback_lock(&mut engine, k, ts, flashback_start_ts),
+            must_prewrite_flashback_key(&mut engine, k, 2, flashback_start_ts),
             1
         );
-        // Only lock `k` since the value of `k` has already existed.
+    }
+
+    #[test]
+    fn test_prewrite_with_special_key() {
+        let mut engine = TestEngineBuilder::new().build().unwrap();
+        let mut ts = TimeStamp::zero();
+        let (k, v1, v2) = (b"k", b"v1", b"v2");
+        must_prewrite_put(&mut engine, k, v1, k, *ts.incr());
+        must_commit(&mut engine, k, ts, *ts.incr());
+        must_prewrite_put(&mut engine, k, v2, k, *ts.incr());
+        must_commit(&mut engine, k, ts, *ts.incr());
+        must_get(&mut engine, k, ts, v2);
+
+        let prewrite_key = b"flashback_2pc";
```

Comment:
I suggest using a more intuitive key like "j" to indicate it's before the "k".

```suggestion
        let start_key = b"j";
```